src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerDifferenceService.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services.diff
+
+import org.apache.commons.lang3.builder.Diff
+import org.apache.commons.lang3.builder.DiffBuilder
+import org.apache.commons.lang3.builder.DiffResult
+import org.apache.commons.lang3.builder.ToStringStyle
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import kotlin.reflect.full.findAnnotations
+
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class DiffableProperty(val type: PropertyType)
+
+enum class PropertyType {
+  IDENTIFIERS, PERSONAL_DETAILS, STATUS, LOCATION, SENTENCE, RESTRICTED_PATIENT
+}
+
+data class Difference(val property: String, val propertyType: PropertyType, val oldValue: Any, val newValue: Any)
+
+fun getDifferencesByPropertyType(prisoner: Prisoner, other: Prisoner): Map<PropertyType, List<Difference>> =
```

Comment:
Our main API that is agnostic to the difference engine used. See the tests for examples of usage.